typing/ctype.ml

```diff
@@ -1247,7 +1247,8 @@ let get_new_abstract_name s =
   if index = 0 && s <> "" && s.[String.length s - 1] <> '$' then s else
   Printf.sprintf "%s%d" s index
 
-let new_declaration expansion_scope manifest =
+let new_local_type ?(loc = Location.none)
```

Comment:
This is basically a smart constructor, and I expect smart constructors to be smart, i.e. provide sensible defaults.
This function is only used in a small number of occurrences, and I think having defaults here avoids cluttering.

Reply:
But `Location.none` is not a *smart* or *sensible* default for an optional location argument, it is the guarantee that some user down the line will get a worse experience because the location is not shown.

Reply:
I don't want to hold the patch up over a triviality, but I pretty strongly disagree with this. Using `Btype.lowest_level` for the expansion scope would be a subtle soundness bug at some of the uses of this function, and using `Location.none` at some other uses would cause us to fail to produce some usage warnings. It's much more important to avoid those bugs than to avoid cluttering the call sites of a rarely used function. I'm less worried about the `manifest` parameter having a default since that one should be much harder to forget.

Reply:
For `Location.none`, I beg to disagree. This is the only possible default. This function is called from places where no location is available, and it would make no sense to make that more explicit. We're not talking of a 3rd party API but of an internal function of the compiler.
For `expansion_scope`, you have a point, and looking back at the code it seems actually needlessly complicated.
I.e., the `expansion_scope` is used only when one expands an abbreviation, so it makes no sense to give it if there is no manifest. So I have changed the arguments to `?loc` and `?manifest_and_scope`, which make much more sense. The testsuite goes through, demonstrating that setting it explicitly was useless.

Reply:
I disagree about `loc`, but that one worries me much less than `expansion_scope`, so I'm happy for it to be merged in its current state.